src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -547,41 +546,124 @@ private void SetDirty(TimeSpan? maxDelay)
-                            client,
-                            cancellationToken)
-                            .ConfigureAwait(false);
-
-                        watchedSettings = await LoadKeyValuesRegisteredForRefresh(
-                            client,
-                            data,
-                            cancellationToken)
-                            .ConfigureAwait(false);
-
-                        watchedSettings = UpdateWatchedKeyValueCollections(watchedSettings, data);
-                    },
-                    cancellationToken)
-                    .ConfigureAwait(false);
+                        await Task.Delay(FailOverConstants.MinStartupBackoffDuration.CalculateBackoffDuration(FailOverConstants.MaxBackoffDuration, attempts), cancellationToken).ConfigureAwait(false);
+                    }
+                    catch (OperationCanceledException)
+                    {
+                        throw new TimeoutException($"The provider timed out while attempting to load.",
+                            new AggregateException(startupExceptions));
+                    }
+                }
             }
             catch (Exception exception) when (
                 ignoreFailures &&
                 (exception is RequestFailedException ||
+                exception is KeyVaultReferenceException ||
+                exception is TimeoutException ||
                 exception is OperationCanceledException ||
                 ((exception as AggregateException)?.InnerExceptions?.Any(e =>
                     e is RequestFailedException ||
                     e is OperationCanceledException) ?? false)))
             { }
+        }
+
+        private async Task<bool> TryInitializeAsync(IEnumerable<ConfigurationClient> clients, List<Exception> startupExceptions, CancellationToken cancellationToken = default)
+        {
+            try
+            {
+                await InitializeAsync(clients, cancellationToken).ConfigureAwait(false);
+            }
+            catch (OperationCanceledException)
+            {
+                if (cancellationToken.IsCancellationRequested)
+                {
+                    return false;
+                }
+
+                throw;
+            }
+            catch (RequestFailedException exception)
+            {
+                if (IsFailOverable(exception))
+                {
+                    startupExceptions.Add(exception);
+
+                    return false;
+                }
+
+                throw;
+            }
+            catch (AggregateException exception)
+            {
+                if (exception.InnerExceptions?.Any(e => e is OperationCanceledException) ?? false)
+                {
+                    if (cancellationToken.IsCancellationRequested)
+                    {
+                        return false;
+                    }
+                }
+
+                if (exception.InnerExceptions?.Any(e => e is RequestFailedException) ?? false)
```

Comment:
```suggestion
                if (exception.InnerExceptions != null &&
                    exception.InnerExceptions.Any(e => e is RequestFailedException))
```

A suggestion, I feel the ?? false is a bit weird to read.